haystack/core/pipeline/pipeline.py

```diff
@@ -197,9 +197,9 @@ def add_component(self, name: str, instance: Component) -> None:
             )
             raise PipelineError(msg)
 
-        # Create the component's input and output sockets
-        input_sockets = getattr(instance, "__haystack_input__", {})
-        output_sockets = getattr(instance, "__haystack_output__", {})
+        # Get the sockets of the component, we're completely sure the fields exist so we ignore the type error
+        input_sockets = instance.inputs._sockets  # type: ignore[attr-defined]
+        output_sockets = instance.outputs._sockets  # type: ignore[attr-defined]
```

Comment:
do we even need these two assignments at this point? Can't we pass `instance.outputs._sockets` directly to `self.graph.add_node(...`?